### PR TITLE
Common component resource settings

### PR DESCRIPTION
**Pull request by thezultimate**

### CODE REVIEW

File: pipeline-runner/steps/promotion.go [hunk 1]
```diff
@@ -185,6 +186,11 @@ func mergeWithRadixApplication(radixConfig *v1.RadixApplication, radixDeployment
 				radixDeployment.Spec.Components[index].EnvironmentVariables[variableKey] = variableValue
 			}
 		}
+
+		// Append common resources settings if currently empty
+		if reflect.DeepEqual(radixDeployment.Spec.Components[index].Resources, v1.ResourceRequirements{}) {
```
Comment (keaaa) [inline pipeline-runner/steps/promotion.go:191]: For promotion will Component.Resource take precedence over Component.Environemt[x].Resource? shouldn't it be other way around?

File: pipeline-runner/steps/promotion_test.go [hunk 2]
```diff
@@ -133,6 +133,13 @@ func TestPromote_PromoteToOtherEnvironment_NewStateIsExpected(t *testing.T) {
 							WithName("app").
 							WithCommonEnvironmentVariable("DB_TYPE", "mysql").
 							WithCommonEnvironmentVariable("DB_NAME", "my-db").
+							WithCommonResource(map[string]string{
+								"memory": "64Mi",
+								"cpu":    "250m",
+							}, map[string]string{
+								"memory": "128Mi",
+								"cpu":    "500m",
+							}).
```
Comment (keaaa) [inline pipeline-runner/steps/promotion_test.go:142]: maybe also test the precedence

File: pkg/apis/deployment/radixcomponent.go [hunk 3]
```diff
@@ -45,6 +46,11 @@ func getRadixComponentsForEnv(radixApplication *v1.RadixApplication, containerRe
 			}
 		}
 
+		// Append common resources settings if currently empty
+		if reflect.DeepEqual(resources, v1.ResourceRequirements{}) {
```
Comment (keaaa) [inline pkg/apis/deployment/radixcomponent.go:50]: same here - environmentspecificconfig should take precedence over common component settings

File: pkg/apis/deployment/radixcomponent_test.go [hunk 4]
```diff
@@ -227,6 +227,50 @@ func TestGetRadixComponentsForEnv_CommonEnvironmentVariables_With_Override(t *te
 	assert.Equal(t, "environment_common_2_dev_override", deployComponentDev[1].EnvironmentVariables["ENV_COMMON_2"])
 }
 
+func TestGetRadixComponentsForEnv_CommonResources(t *testing.T) {
```
Comment (keaaa) [inline pkg/apis/deployment/radixcomponent_test.go:230]: test precedence

Comment (thezultimate) [reply pkg/apis/deployment/radixcomponent_test.go:230]: Precedence is tested in line 260-263.

File: pipeline-runner/steps/promotion.go (hunk 1 above)
Comment (thezultimate) [reply pipeline-runner/steps/promotion.go:191]: No, Component.Environemt[x].Resource takes precedence over Component.Resource.

The logic is:
(1) Set value to Component.Environemt[x].Resource
(2) If (1) is empty, set value to Component.Resource

We use the standard DeepEqual function to check if a struct is empty.

File: pipeline-runner/steps/promotion_test.go (hunk 2 above)
Comment (thezultimate) [reply pipeline-runner/steps/promotion_test.go:142]: I will update the test.

**Pull request merged**